lpa-submissions-e2e-tests/cypress/integration/magicLink/magicLink.js

```diff
@@ -0,0 +1,178 @@
+import { And, Given, Then, When } from 'cypress-cucumber-preprocessor/steps';
+import verifyPage from '../../support/common/verifyPage';
+import verifyNotificationBanner from '../../support/common/verifyNotificationBanner';
+import validateErrorMessage from '../../support/common/validateErrorMessage';
+import goToPage from '../../support/common/goToPage';
+import clickLink from '../../support/common/clickLink';
+import clickSubmit from '../../support/common/clickSubmitButton';
+import hasLink from '../../support/common/hasLink';
+import inputEmailAddress from '../../support/magic-link/inputEmailAddress';
+import authenticateLPA from '../../support/magic-link/authenticateLPA';
+import getMagicLink from '../../support/magic-link/getMagicLink';
+import createAuthToken from '../../support/magic-link/createAuthToken';
+
+const lpaCode = 'E69999999';
+const enterEmailAddressLink = 'authentication/your-email';
+const confirmEmailAddressLink = 'authentication/confirm-email';
+const questionnairePage = 'task-list';
+const lpaName = 'System Test Borough Council';
+
+Given('LPA wants to access a questionnaire', () => {
+  // nothing to do here
+});
+
+Given('the LPA is on the confirm your email page', () => {
+  goToPage(confirmEmailAddressLink, lpaCode);
+});
+
+Given('LPA user receives a magic link for accessing the questionnaire', () => {
+  goToPage(questionnairePage);
+  inputEmailAddress();
+  clickSubmit();
+});
+
+And('is trying to access the LPA questionnaire', () => {
+  // nothing to do here
+});
+
+Given('access to the questionnaire is requested', () => {
+  goToPage(questionnairePage);
+});
+
+Given('LPA user wants to have the magic link resent to them', () => {
+  goToPage(confirmEmailAddressLink, lpaCode);
+});
+
+Given('the session has timed out', () => {
+  const tokenExpirationTime = new Date();
+  tokenExpirationTime.setHours(tokenExpirationTime.getHours() - 4); // 4h ago
+  const expiredAuthToken = createAuthToken(tokenExpirationTime.getTime());
+
+  cy.setCookie(Cypress.env('AUTH_COOKIE_NAME'), expiredAuthToken);
+});
+
+Given('LPA Planning Officer wants to complete a questionnaire', () => {
+  // nothing to do here
+});
+
+Given('the LPA Planning Officer has not been authenticated', () => {
+  cy.clearCookies();
+});
+
+Given('the LPA Planning Officer is authenticated', () => {
+  authenticateLPA();
+});
+
+When('they click on the link in the start email', () => {
+  goToPage(questionnairePage);
+});
+
+When('the email address does not match the domain of the LPA from the appeal', () => {
+  inputEmailAddress('test@test.gov.uk');
+  clickSubmit();
+});
+
+When('an email address is not provided', () => {
+  inputEmailAddress('');
+  clickSubmit();
+});
+
+When('the email address provided in not in the correct format', () => {
+  inputEmailAddress('invalidEmailAddress');
```

Comment:
Correct, it is not required for the cypress test to check whether or not the entered email adheres to the email format.

Reply:
@swdpcomputing _When_ is a condition rather than validation. Here we are setting up the data in the expected format (non email format in this context).